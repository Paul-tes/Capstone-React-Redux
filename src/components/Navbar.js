import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../logo/planet.png';
import './style/Navbar.css';

export default function Navbar() {
  return (
    <div className="Navbar">
      <div className="left-nav">
        <img src={logo} alt="logo" width={50} height={50} />
        <h1>Space Travelers&apos; Hub</h1>
      </div>
      <div className="right-nav">
        <NavLink to="/">Rocket</NavLink>
        <NavLink to="missions">Missions</NavLink>
        <div className="vertical-line">.</div>
        <NavLink to="profile">My Profile</NavLink>
      </div>
    </div>
  );
}
