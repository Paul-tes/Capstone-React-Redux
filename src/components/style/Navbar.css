.Navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 5px 30px;
  box-shadow: 0 10px 7px -4px rgba(0, 0, 0, 0.1);
}

.left-nav,
.right-nav {
  display: flex;
  align-items: center;
  gap: 20px;
}

.vertical-line {
  border-left: 1px solid #007bff;
  height: 20px;
}

a:visited,
a:active,
a:link {
  text-decoration: none;
  font-size: 18px;
  color: #007bff;
}

.active {
  color: #007bff;
  text-decoration: underline;
  font-weight: bold;
}
