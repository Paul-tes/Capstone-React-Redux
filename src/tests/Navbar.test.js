import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

test('renders the navbar component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders Rocket in navigation', async () => {
  render(
    <Navbar />,
  );
  const linkElement = screen.getByText(/Rocket/i);
  expect(linkElement).toBeInTheDocument();
});
