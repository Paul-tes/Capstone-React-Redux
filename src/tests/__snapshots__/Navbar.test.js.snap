// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the navbar component snapshot 1`] = `
<DocumentFragment>
  <div
    class="Navbar"
  >
    <div
      class="left-nav"
    >
      <img
        alt="logo"
        height="50"
        src="planet.png"
        width="50"
      />
      <h1>
        Space Travelers' Hub
      </h1>
    </div>
    <div
      class="right-nav"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Rocket
      </a>
      <a
        class=""
        href="/missions"
      >
        Missions
      </a>
      <div
        class="vertical-line"
      />
      <a
        class=""
        href="/profile"
      >
        My Profile
      </a>
    </div>
  </div>
</DocumentFragment>
`;
